merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/generator/MissionModelGenerator.java

```diff
@@ -646,6 +658,100 @@ public JavaFile generateActivityTypes(final MissionModelRecord missionModel) {
         .build();
   }
 
+  public Pair<JavaFile, List<TypeRule>> generateAutoValueMappers(
+      final MissionModelRecord missionModel,
+      final Iterable<TypeElement> recordTypes) {
+    final var typeRules = new ArrayList<TypeRule>();
+
+    final var typeName = missionModel.getAutoValueMappersName();
+
+    final var builder =
+        TypeSpec
+            .classBuilder(typeName)
+            .addAnnotation(
+                AnnotationSpec
+                    .builder(javax.annotation.processing.Generated.class)
+                    .addMember("value", "$S", MissionModelProcessor.class.getCanonicalName())
+                    .build())
+            .addModifiers(Modifier.PUBLIC, Modifier.FINAL);
+
+    record ComponentMapperNamePair(String componentName, String mapperName) {}
+    for (final var record : recordTypes) {
+      if (record.getKind() != ElementKind.RECORD) {
+        messager().printError("@%s is only allowed on records".formatted(AutoValueMapper.class.getSimpleName()), record);
```

Comment:
Good edge-case to check for! This looks like something that should likely be in the mission model parser logic instead of the code generator though.

Reply:
I've moved this to the parser, as requested :+1: